### PR TITLE
Make all list commands use FormatWritter interface

### PR DESCRIPTION
**Pull request by difince**
- DeviceService, Event, Intervals, Notifications, DeviceProfile, Reading
and  Subscription lists commands use FormatWritter interface
- Status command user FormatWritter interface
(The result table have table header row
Base service url is also printed)
- EmptyInterface is removed

Based on PR:  https://github.com/edgexfoundry-holding/edgex-cli/pull/212

### CODE REVIEW

File: cmd/deviceservice/list/list.go [hunk 1]
```diff
@@ -15,49 +15,43 @@
...
-	"text/tabwriter"
-
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
+const dsTemplete = "Service ID\tService Name\tOperating State\tAdmin State\tAddressable Name\tCreated\n" +
```
Comment (mkbhanda) [inline cmd/deviceservice/list/list.go:30]: minor detail — s/templete/template. https://www.dictionary.com/browse/template

File: cmd/event/list/list.go [hunk 2]
```diff
@@ -29,9 +28,13 @@ import (
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
+const eventTemplete = "Event ID\tDevice\tOrigin\tCreated\tModified\n" +
```
Comment (mkbhanda) [inline cmd/event/list/list.go:33]: s/Templete/Template

File: cmd/interval/list/list.go [hunk 3]
```diff
@@ -46,23 +48,8 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
-			interval.Name,
-			interval.Start,
-			interval.End,
-			interval.Frequency,
-			interval.Cron,
-			interval.RunOnce,
-		)
-	}
-	w.Flush()
+	formatter := formatters.NewFormatter("commandList", intervalTemplete, nil)
```
Comment (mkbhanda) [inline cmd/interval/list/list.go]: "intervalList" instead of commandList ?

Comment (mkbhanda) [reply cmd/interval/list/list.go]: could we just fix it to some default given we have only one instance at any time in cli/

Comment (difince) [reply cmd/interval/list/list.go]: Done - one default name is used

File: cmd/event/list/list.go (hunk 2 above)
Comment (mkbhanda) [inline cmd/event/list/list.go:33]: s/Templete/Template

File: cmd/interval/list/list.go [hunk 4]
```diff
@@ -16,21 +16,23 @@ package list
...
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/scheduler"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
+const intervalTemplete = "Interval ID\tName\tStart\tEnd\tFrequency\tCron\tRunOnce\n" +
```
Comment (mkbhanda) [inline cmd/interval/list/list.go:31]: s/Templete/Template

File: cmd/interval/list/list.go [hunk 5]
```diff
@@ -46,23 +48,8 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
-			interval.Name,
-			interval.Start,
-			interval.End,
-			interval.Frequency,
-			interval.Cron,
-			interval.RunOnce,
-		)
-	}
-	w.Flush()
+	formatter := formatters.NewFormatter(intervalTemplete, nil)
```
Comment (mkbhanda) [inline cmd/interval/list/list.go:51]: s/Templete/Template

File: cmd/notification/list/list.go [hunk 6]
```diff
@@ -40,25 +38,18 @@ var labels string
 var sender string
 var onlyNew bool
 
+const notificationTemplete = "Notification ID\tSlug\tSender\tStatus\tSeverity\tCategory\tContent\tLabels\tCreated\tModified\n" +
```
Comment (mkbhanda) [inline cmd/notification/list/list.go:41]: s/Templete/Template

File: cmd/notification/list/list.go [hunk 7]
```diff
@@ -120,24 +111,7 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
-			notification.Severity,
-			notification.Category,
-			notification.Content,
-			notification.Labels,
-			utils.DisplayDuration(notification.Created),
-			utils.DisplayDuration(notification.Modified),
-		)
-	}
-	w.Flush()
+	formatter := formatters.NewFormatter(notificationTemplete, template.FuncMap{"DisplayDuration": utils.DisplayDuration})
```
Comment (mkbhanda) [inline cmd/notification/list/list.go:114]: s/Templete/Template

File: cmd/profile/list/list.go [hunk 8]
```diff
@@ -16,57 +16,48 @@ package list
...
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
+const profileTemplete = "Profile ID\tProfile Name\tManufacturer\tModel\tCreated\tModified\n" +
```
Comment (mkbhanda) [inline cmd/profile/list/list.go:32]: s/Templete/Template

File: cmd/reading/list/list.go [hunk 9]
```diff
@@ -15,89 +15,62 @@
...
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
 var limit int32
 
+const readingTemplete = "Reading ID\tName\tDevice\tOrigin\tValue\tCreated\tModified\tPushed\n" +
```
Comment (mkbhanda) [inline cmd/reading/list/list.go:34]: s/Templete/Template

File: cmd/reading/list/list.go [hunk 10]
```diff
@@ -15,89 +15,62 @@
...
+		url = config.Conf.Clients["CoreData"].Url() + clients.ApiReadingRoute
+	}
+	var readings []models.Reading
+
+	err = request.Get(url, &readings)
+	if err != nil {
+		return
+	}
+
+	formatter := formatters.NewFormatter(readingTemplete, template.FuncMap{"DisplayDuration": utils.DisplayDuration})
```
Comment (mkbhanda) [inline cmd/reading/list/list.go:73]: s/Templete/Template

File: cmd/status/status.go [hunk 11]
```diff
@@ -39,27 +48,30 @@ This command pings each edgex microservice and prints their status.
...
 					if string(data) == "pong" {
-						fmt.Fprintf(w, "%s \t OK\n", clientName)
+						clientStatuses = append(clientStatuses, clientStatus{clientName,client.Url(), OK})
 					}
 					resp.Body.Close()
 				}
 			}
-			w.Flush()
-
+			formatters.NewHtmlTempleteFormatter(statusTemplate, nil).Write(clientStatuses)
```
Comment (mkbhanda) [inline cmd/status/status.go:67]: s/Templete/Template

File: cmd/subscription/list/list.go [hunk 12]
```diff
@@ -15,51 +15,41 @@
...
+
+func listHandler(cmd *cobra.Command, args []string) (err error) {
+	url := config.Conf.Clients["Notification"].Url() + clients.ApiSubscriptionRoute
+	var subscriptions []models.Subscription
+	err = request.Get(url, &subscriptions)
+	if err != nil {
+		return
+	}
+
+	formatter := formatters.NewFormatter(subscriptionTemplete, nil)
```
Comment (mkbhanda) [inline cmd/subscription/list/list.go:52]: s/Templete/Template

File: pkg/formatters/interface.go [hunk 13]
```diff
@@ -22,9 +22,9 @@ type FormatWriter interface {
...
 
-func NewFormatter (name string, template string, funcMaps template.FuncMap) FormatWriter {
+func NewFormatter (template string, funcMaps template.FuncMap) FormatWriter {
 	if viper.GetBool("verbose") {
 		return &JsonFormatter{}
 	}
-	return NewHtmlTempleteFormatter(name, template, funcMaps)
-}
\ No newline at end of file
+	return NewHtmlTempleteFormatter(template, funcMaps)
```
Comment (mkbhanda) [inline pkg/formatters/interface.go:29]: s/Templete/Template

File: cmd/event/list/list.go (hunk 2 above)
Comment (mkbhanda) [reply cmd/event/list/list.go:33]: :-( I still see Templete .. 
"const eventTemplete = "Event"

File: cmd/event/list/list.go [hunk 14]
```diff
@@ -66,21 +69,8 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
-		fmt.Fprintf(w, "%s\t%s\t%v\t%v\t%s\t\n",
-			event.ID,
-			event.Device,
-			event.Origin,
-			utils.DisplayDuration(event.Created),
-			utils.DisplayDuration(event.Modified),
-		)
-	}
-	w.Flush()
+	formatter := formatters.NewFormatter(eventTemplete, template.FuncMap{"DisplayDuration": utils.DisplayDuration})
```
Comment (mkbhanda) [inline cmd/event/list/list.go:72]: eventTemplete

File: cmd/notification/list/list.go (hunk 7 above)
Comment (difince) [reply cmd/notification/list/list.go]: done

File: cmd/deviceservice/list/list.go (hunk 1 above)
Comment (difince) [reply cmd/deviceservice/list/list.go]: done

File: cmd/event/list/list.go (hunk 2 above)
Comment (difince) [reply cmd/event/list/list.go]: done

File: cmd/reading/list/list.go (hunk 10 above)
Comment (difince) [reply cmd/reading/list/list.go]: done

File: cmd/reading/list/list.go (hunk 9 above)
Comment (difince) [reply cmd/reading/list/list.go]: done

File: cmd/event/list/list.go (hunk 14 above)
Comment (difince) [reply cmd/event/list/list.go]: done

**Comment by mkbhanda** [comment]
Thank you!!

**Pull request merged**